gluten-core/src/main/scala/io/glutenproject/extension/ExpandFallbackPolicy.scala

```diff
@@ -151,10 +157,7 @@ case class ExpandFallbackPolicy(isAdaptiveContext: Boolean, originalPlan: SparkP
               // we make the stage fall back.
               case tableCache if InMemoryTableScanHelper.isGlutenTableCache(tableCache) =>
                 stageFallbackCost = stageFallbackCost + 1
-              // For other case, table cache will save internal RowToColumnar if we make
-              // the stage fall back.
               case _ =>
-                stageFallbackCost = stageFallbackCost - 1
```

Comment:
@PHILO-HE I move it to the `countFallback` and add a config to control if we consider `RowToColumnar` as the fallback operator. Before, it's possbile that the `fallbackCost` is a negative number which looks a bit confused. After, we just count it to the fallback number if ignoreRowToColumnar is false.

Reply:
I think a negative fallback cost may make sense as it votes for falling back the stage, although the name is confusing.

Reply:
The point is that it's possible the gluten plan contains other RowToColumnar operators and we should handle it like fallback cost, so I move it to countFallback. If we want to keep a negative fallback cost here, we should change it to count the RowToColumnar operators instead of the row-based exchange. Does it sound good ?

Reply:
In the initial design, we just considered fallback times which is measured by counting `ColumanrToRow` as you know. In Gluten's current code, as `Row2Columnar` is considered, negative fallback cost looks confusing indeed. But it also looks a bit confusing if we count it as fallback. Maybe, we can use some different concepts to make the code clear, e.g., transition cost.

For a given stage transformed by Gluten, I think we can calculate `transitionCost` by counting all `ColumnarToRow` & `RowToColumnar`. Then,  we can calculate `StageFallbackTransitionCost` by counting all `ColumnarToRow` for the fallback stage. Here, the usage of `ColumnarToRow` comes from: 1) last stage's columnar output; 2) vanilla spark's vectorized reader (use a config to control whether to count this?).

By comparing `transitionCost` - `StageFallbackTransitionCost` with user-specified threshold, we can let gluten fall back the stage or not.

It is basically equivalent to Gluten's current approach of evaluating fallback count & fallback cost. But I think it is more clear logically. Does it make sense?

Reply:
thank you @PHILO-HE  agree, I will refine the related code and naming.